app/logic/belongings.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import annotations
+
+
+class Armor:
+    def __init__(
+            self,
+            part: str,
+            protection: int
+    ) -> None:
+        self.part = part
+        self.protection = protection
+
+
+class Weapon:
+    def __init__(
+            self,
+            name: str,
+            power: int
+    ) -> None:
+        self.name = name
+        self.power = power
+
+    @staticmethod
+    def initialize(
+            weapon: dict
+    ) -> Weapon:
+        return Weapon(weapon["name"], weapon["power"])
+
+
+class Potion:
+    def __init__(
+            self,
+            name: str,
+            effect: Effect
+    ) -> None:
+        self.name = name
+        self.effect = effect
+
+    @staticmethod
+    def check_effect(effect_dict: dict) -> Effect:
+        result = Effect.find_effect(Effect.make_key(effect_dict))
+        if result:
+            return result
+        return Effect(
+            hp=effect_dict.get("hp") or 0,
+            power=effect_dict.get("power") or 0,
+            protection=effect_dict.get("protection") or 0
+        )
+
+    @staticmethod
+    def initialize(
+            potion: dict | list
```

Comment:
add additional typehint for list inner objects like list[YOUR_VALUE_TYPE]
for example: list that contains string elements has typehint list[str]